packages/orchestration/src/exos/ibc-packet.js

```diff
@@ -168,14 +182,26 @@ export const prepareIBCReplyKit = (zone, vowTools) => {
   const ibcWatcher = zone.exo(
     'ibcResultWatcher',
     M.interface('processIBCWatcher', {
-      onFulfilled: M.call(M.record(), M.record()).returns(Vow$(M.string())),
+      onFulfilled: M.call(
+        M.splitRecord({ event: M.string() }, { acknowledgement: M.string() }),
+        M.record(),
+      ).returns(Vow$(M.string())),
     }),
     {
-      onFulfilled({ event, acknowledgement }, { opName = 'unknown' }) {
-        assertAllDefined({ event, acknowledgement });
+      /**
+       * @param {IBCEvent<'acknowledgementPacket' | 'timeoutPacket'>} ibcEvent
+       * @param {PacketOptions} ctx
+       */
+      onFulfilled(ibcEvent, { opName = 'unknown' }) {
+        const { event } = ibcEvent;
         switch (event) {
-          case 'acknowledgementPacket':
+          case 'acknowledgementPacket': {
+            const { acknowledgement } =
+              /** @type {IBCEvent<'acknowledgementPacket'>} */ (ibcEvent);
```

Comment:
Shouldn't need a cast if `event` isn't destructured from `ibcEvent`:

```js
switch (ibcEvent.event) {
  case 'acknowledgementPacket': {
    const { acknowledgement } = ibcEvent;
    ...
```

Reply:
Yep this goes away once we use `IBCEvent<'acknowledgementPacket'> | IBCEvent<'timeoutPacket'>` instead of `IBCEvent<'acknowledgementPacket' | 'timeoutPacket'>`